qiskit/primitives/base/base_estimator.py

```diff
@@ -254,3 +346,44 @@ def parameters(self) -> tuple[ParameterView, ...]:
             Parameters, where ``parameters[i][j]`` is the j-th parameter of the i-th circuit.
         """
         return tuple(self._parameters)
+
+
+BaseEstimator = BaseEstimatorV1
+
+
+class BaseEstimatorV2(BasePrimitiveV2, Generic[T]):
+    """Estimator base class version 2.
+
+    Estimator estimates expectation values of quantum circuits and observables.
+    """
+
+    def __init__(self, options: Optional[BasePrimitiveOptionsLike]):
+        super().__init__(options=options)
+
+    def run(self, tasks: EstimatorTaskLike | Iterable[EstimatorTaskLike]) -> T:
+        """Run the tasks of the estimation of expectation value(s).
+
+        Args:
+            tasks: a tasklike object. Typically, list of tuple
+                ``(QuantumCircuit, observables, parameter_values)``
+
+        Returns:
+            The job object of Estimator's Result.
+        """
+        if isinstance(tasks, EstimatorTask):
+            tasks = [tasks]
+        elif isinstance(tasks, tuple) and isinstance(tasks[0], QuantumCircuit):
+            tasks = [EstimatorTask.coerce(tasks)]
+        elif isinstance(tasks, Iterable):
+            tasks = [EstimatorTask.coerce(task) for task in tasks]
+        else:
+            raise TypeError(f"Unsupported type {type(tasks)} is given.")
+
+        for task in tasks:
+            task.validate()
+
+        return self._run(tasks)
+
+    @abstractmethod
+    def _run(self, tasks: list[EstimatorTask]) -> T:
```

Comment:
Can be removed if we make `run` the abstract method and leave validation as a utility function

Reply:
I'm fine, but your proposal is to delegate the calling of the validation to the implementor of subclasses, is that correct? I am concerned that the subclass implementer may forget to call it.

Reply:
Now `run` is the abstract method